spec/uap_clj/common_spec.clj

```diff
@@ -1,3 +1,27 @@
-(ns uap-clj.common-spec)
+(ns uap-clj.common-spec
+  (:require [clojure.java.io :as io :refer [resource]]
+            [clj-yaml.core :refer [parse-string]])
+  (:import [org.yaml.snakeyaml LoaderOptions]))
 
-(def unknown-ua "Unknown new useragent in the wild/v0.1.0")
+(def ^:const unknown-ua "Unknown new useragent in the wild/v0.1.0")
+
+(defn default-codepoint-limit
+  []
+  (let [options (LoaderOptions.)]
+    (.getCodePointLimit options)))
+
+(defn snakeyaml-loader-options
+  ^LoaderOptions [n]
+  (let [options (LoaderOptions.)]
+    (.setCodePointLimit options n)
+    options))
+
+(defn load-fixture
+  [f]
+  (let [raw-file (slurp (io/resource f))
+        file-size (count raw-file)]
+    (if (>= file-size (default-codepoint-limit))
+      (:test_cases
+        (with-redefs [clj-yaml.core/default-loader-options #(snakeyaml-loader-options file-size)]
+          (parse-string raw-file)))
+      (:test_cases (parse-string raw-file)))))
```

Comment:
From what I gather, you only read local files here and don't need to fetch any resources via a URL, so to potentially reduce memory pressure during builds you could use the file and stream API. I.e.
```clojure
(let [file (io/file f)
      file-size (.length file)]
    (if (>= file-size (default-codepoint-limit))
      (:test_cases
        (with-redefs [clj-yaml.core/default-loader-options #(snakeyaml-loader-options file-size)]
          (parse-stream file)))
      (:test_cases (parse-stream file)))))
```
Of course, if that actually safes memory depends on the SnakeYAML implementation, but I think it's preferable to use lazy APIs where possible.

If working in tests the same idea could probably applied to https://github.com/ua-parser/uap-clj/blob/62ce2aeca0c33444537a5d357a3a24ecce69d0ba/src/uap_clj/common.clj#L7 (even though memory-wise it should only matter during startup, since it's static code).

Reply:
@bitti I merged before your last commentary, but will definitely incorporate this feedback in the next revision, thanks.